### PR TITLE
prov/sm2: Move functions to header, and make them inline

### PR DESCRIPTION
**Pull request by a-szegel**
Break out copy/pasted logic into sm2_pop_xfer_entry(), and move it to the header file so that it can be used by MSG/ATOMICS/RMA API's.

Move sm2_generic_format() to the header so it can be used by MSG/ATOMICS/RMA apis.

### CODE REVIEW

File: prov/sm2/src/sm2_ep.c [hunk 1]
```diff
@@ -271,22 +257,13 @@ static ssize_t sm2_do_inject(struct sm2_ep *ep, struct sm2_region *peer_smr,
...
-	self_region = sm2_mmap_ep_region(map, ep->gid);
-
-	if (smr_freestack_isempty(sm2_freestack(self_region))) {
-		sm2_progress_recv(ep);
-		if (smr_freestack_isempty(sm2_freestack(self_region)))
-			return -FI_EAGAIN;
-	}
-
-	xfer_entry = smr_freestack_pop(sm2_freestack(self_region));
+	ret = sm2_pop_xfer_entry(ep, &xfer_entry);
```
Comment (sunkuamzn) [inline prov/sm2/src/sm2_ep.c:264]: We have the same code in CMA and IPC as well. I will add that to the CMA and IPC commits after you merge this.

**Review by sunkuamzn — APPROVED** [review]

**Comment by sunkuamzn** [comment]
clang format check failing

**Review by lrbison — APPROVED** [review]

**Comment by lrbison** [comment]
@aingerson is the Intel CI failure real?

**Review by aingerson — APPROVED** [review]

**Comment by aingerson** [comment]
@lrbison No, our cluster is having issues again. All PRs will fail our CI until they are resolved :/

**Comment by lrbison** [comment]
Thank you for checking!

**Pull request merged**